Analytics/models/prediction_results.py

```diff
@@ -0,0 +1,118 @@
+''' Data table to store the results of get_prediction requests '''
+
+from datetime import datetime
+import json
+
+from sqlalchemy.dialects.postgresql import JSON
+from sqlalchemy.exc import IntegrityError
+import logging
+
+from db import db
+
+logging.basicConfig(level='INFO')
+logger = logging.getLogger(__name__)
+
+CACHE_SIZE = 20
+
+
+class PredictionResults(db.Model):
+    __tablename__ = 'predictionresults'
+
+    id = db.Column(db.Integer, primary_key=True)
+    forcasting_engine = db.Column(db.String(50))
+    mean_absolute_percentage_error = db.Column(db.Float)
+    attribute_table = db.Column(db.String(150), nullable=False)
+    sensor_id = db.Column(db.String(150), nullable=False)
+    num_predictions = db.Column(db.Integer, nullable=False)
+    result = db.Column(JSON, nullable=False)
+    timestamp = db.Column(db.DateTime, nullable=False)
+
+    def __init__(self, eng: str, mape: float, attribute_table: str,
+                 sensor_id: str,  num_predictions: int, result: JSON,
+                 timestamp: datetime = None):
+        """
+        Initialise the RequestPrediction object instance
+        :param eng: the name of the forecasting engine used for predictions
+        :param mape: the mean absolute percent error over all prediction values
+        :param attribute_table: the attribute table whose entries were used
+        in the prediction calculations
+        :param sensor_id: the ID of the sensor used during prediction
+        calculations
+        :param num_predictions: the number of predictions that were calculated
+        :param result: the prediction list returned from get_predictions
+        :param timestamp: time stamp of when the result was stored
+        """
+        self.forcasting_engine = eng
+        self.mean_absolute_percentage_error = mape
+        self.attribute_table = attribute_table
+        self.sensor_id = sensor_id
+        self.num_predictions = num_predictions
+        self.result = result
+
+        if timestamp is None:
+            timestamp = datetime.utcnow()
```

Comment:
Replace `timestamp: datetime = None` with `timestamp: datetime = datetime.utcnow()`

Reply:
You also need to remove this line